doc/guidance.md

```diff
@@ -14,13 +14,22 @@ This is about how to use Doxygen to maintain the API documents. There are three
 	- on mac, run "brew install graphviz"
 - goto the root directory of "amazon-freertos" (this will be referenced as $ROOT_AFR_DIR)
 	- cd $ROOT_AFR_DIR
-- create a "tag" folder in "ROOT/doc
```

Comment:
Why remove this?

Reply:
Per feedback from @cobusve. Seems like it was an unnecessary folder. Now tag files just get created in output folder.

Reply:
In the past (a few weeks ago lol), I had to create that folder otherwise I could not generate.